Ginger/GingerCore/ALM/Qtest/ImportFromQtest.cs

```diff
@@ -0,0 +1,1128 @@
+                                    continue;//activity which not originally came from the TC
+                                numOfSeenSteps++;
+
+                                if (numOfSeenSteps >= stepIndx) break;
+                            }
+                            ActivityIdentifiers identOnPlace = (ActivityIdentifiers)tcActivsGroup.ActivitiesIdentifiers[groupIndx];
+                            if (identOnPlace.ActivityGuid != act.Guid)
+                            {
+                                //replace places in group
+                                tcActivsGroup.ActivitiesIdentifiers.Move(groupActIndx, groupIndx);
+                                //replace places in business flow
+                                busFlow.Activities.Move(bfActIndx, startGroupActsIndxInBf + groupIndx);
+                            }
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        Reporter.ToLog(eLogLevel.ERROR, $"Method - {MethodBase.GetCurrentMethod().Name}, Error - {ex.Message}", ex);
+                        //failed to re order the activities to match the tc steps order, not worth breaking the import because of this
+                    }
+                }
+
+                //Add the BF variables (linked variables)
+                if (busVariables.Keys.Count > 0)
+                {
+                    foreach (KeyValuePair<string, string> var in busVariables)
+                    {
+                        //add as String param
+                        VariableString busVar = new VariableString();
+                        busVar.Name = var.Key;
+                        ((VariableString)busVar).InitialStringValue = var.Value;
+                        busFlow.AddVariable(busVar);
+                    }
+                }
+
+                return busFlow;
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to import QC test set and convert it into " + GingerDicser.GetTermResValue(eTermResKey.BusinessFlow), ex);
+                return null;
+            }
+        }
+
+        public static void UpdatedQCTestInBF(ref BusinessFlow busFlow, List<QtestTest> tcsList)
+        {
+            if ((busFlow == null) || (tcsList == null) || (tcsList.Count < 1)) return;
+            Dictionary<string, string> busVariables = new Dictionary<string, string>();
+
+            int startGroupActsIndxInBf = 0;
+            Dictionary<string, int> activityGroupsToRemoveIndexes = new Dictionary<string, int>();
+            foreach (QtestTest tc in tcsList)
+            {
+                var activitiesToRemove = busFlow.Activities.Where(x => x.ActivitiesGroupID == tc.TestName).ToList();
+                foreach (Activity activityToRemove in activitiesToRemove)
+                {
+                    if (startGroupActsIndxInBf < busFlow.Activities.IndexOf(activityToRemove))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)